saleor/static/js/components/variantPicker/VariantPicker.js

```diff
@@ -60,6 +93,11 @@ export default class VariantPicker extends Component {
       selection: Object.assign({}, this.state.selection, { [attrId]: valueId })
     }, () => {
       this.matchVariantFromSelection();
+      let params = {};
+      for (let [id, v] of Object.entries(this.state.selection)) {
+        params[this.idToName[id].name] = this._getAttributeValueSlug(v, this.idToName[id].values);
+      }
+      history.pushState(null, null, '?'+ queryString.stringify(params));
```

Comment:
As mentioned earlier, transformation from current selection to query params could be done without extra structures:
```
matchAttribute = (id) => {
  const { variantAttributes } = this.props;
  const match = variantAttributes.filter(attribute => attribute.pk.toString() === id);
  return match.length > 0 ? match[0] : null;
}

// matchAttributeValue would do the same for attribute values

Object.keys(this.state.selection).forEach(attrId => {
  const attribute = this.matchAttribute(attrId);
  const value = this.matchAttributeValue(attribute, this.state.selection[attrId]);
  if (attr && value) {
     // get both slugs and add them to params
  }
});
```